src/cljs/main/broadfcui/common/components.cljs

```diff
@@ -186,61 +186,75 @@
-                                                :style {:width 100}
-                                                :defaultValue (key entity)
-                                                :onChange (when-let [f (:onSnapshotIdChange props)]
-                                                            #(f (int (common/get-text refs "snapshotId"))))}
-                                               (:snapshots props))
+                 (style/create-identity-select-name {:ref key
+                                                     :data-test-id (config/when-debug "edit-method-config-snapshot-id-select")
+                                                     :style {:width 120}
+                                                     :defaultValue (if redacted-snapshot -1 (key entity))
+                                                     :onChange (when-let [f (:onSnapshotIdChange props)]
+                                                                 #(f (int (common/get-text refs "snapshotId"))))}
+                                                    (:snapshots props)
+                                                    redacted-snapshot)
                  (let [rendered ((or render identity) (key entity))]
                    [:span {:title rendered} rendered]))]])]
        [:div {}
         [:div {:style {:display "flex"}}
          [:div {:style {:flex "1 1 50%" :paddingRight "0.5rem"}}
+          (when redacted?
+            [:div {:style {:fontWeight 500 :paddingBottom "0.25rem"}}
+             (icons/icon {:style {:color (:warning-state style/colors)}} :warning) " Snapshot Redacted"])
           (make-field :namespace "Namespace")
           (make-field :name "Name")
           (make-field :snapshotId "Snapshot ID" :dropdown? true)
           (make-field :entityType "Entity Type")]
-         [:div {:style {:flex "1 1 50%"}}
-          (make-field :createDate "Created" :render common/format-date)
-          (make-field :managers "Owners" :render (partial clojure.string/join ", ") :wrap? true)
-          (make-field :synopsis "Synopsis")]]
-        [:div {:style {:fontWeight 500 :padding "0.5rem 0 0.3rem 0"}}
-         "Documentation:"]
-        (if (string/blank? (:documentation entity))
-          [:div {:style {:fontStyle "italic" :fontSize "90%"}} "No documentation provided"]
-          [:div {:style {:fontSize "90%"}} (:documentation entity)])]))})
+         (when-not redacted?
+           [:div {:style {:flex "1 1 50%"}}
+            (make-field :createDate "Created" :render common/format-date)
+            (make-field :managers "Owners" :render (partial clojure.string/join ", ") :wrap? true)
+            (make-field :synopsis "Synopsis")])]
+        (when-not redacted?
+          [:div {:style {:fontWeight 500 :padding "0.5rem 0 0.3rem 0"}}
+           "Documentation:"]
+          (if (string/blank? (:documentation entity))
```

Comment:
If this is within the `when-not`, it needs to be in a div with the "Documentation:" label.